src/Compiler/Service/ServiceAnalysis.fs

```diff
@@ -451,3 +452,869 @@ module UnusedDeclarations =
+        /// function application, e.g.,
+        ///
+        /// f x
+        ///
+        /// (+) x y
+        let (|HighPrecedenceApp|_|) expr =
+            match expr with
+            | SynExpr.App (isInfix = false; funcExpr = SynExpr.Ident _)
+            | SynExpr.App (isInfix = false; funcExpr = SynExpr.LongIdent(longDotId = SynLongIdent(trivia = [])))
+            | SynExpr.App (isInfix = false; funcExpr = SynExpr.App(isInfix = false)) -> Some HighPrecedenceApp
+            | _ -> None
+
+        module FuncExpr =
+            /// Matches when the given funcExpr is a direct application
+            /// of a symbolic operator, e.g., -, _not_ (~-).
+            let (|SymbolicOperator|_|) funcExpr =
+                match funcExpr with
+                | SynExpr.LongIdent(longDotId = SynLongIdent (trivia = trivia)) ->
+                    trivia
+                    |> List.tryPick (function
+                        | Some (IdentTrivia.OriginalNotation op) -> Some op
+                        | _ -> None)
+                | _ -> None
+
+        /// Represents the infix application of a symbolic binary operator.
+        /// The original notation may include leading dots and trailing characters,
+        /// with the exception of TypeTest, Cons, Upcast, Downcast, BarBar, AmpAmp, and ColonEquals.
+        [<NoComparison; NoEquality>]
+        type InfixOperator =
+            /// :=
+            | ColonEquals
+
+            /// or, ||
+            ///
+            /// Refers to the exact operators or and ||.
+            /// Instances with leading dots or question marks or trailing characters are parsed as Bar instead.
+            | BarBar
+
+            /// &, &&
+            ///
+            /// Refers to the exact operators & and &&.
+            /// Instances with leading dots or question marks or trailing characters are parsed as Amp instead.
+            | AmpAmp
+
+            /// :?>
+            | Downcast
+
+            /// :>
+            | Upcast
+
+            /// =…
+            | Eq
+
+            /// |…
+            | Bar
+
+            /// &…
+            | Amp
+
+            /// $…
+            | Dollar
+
+            /// >…
+            | Greater
+
+            /// <…
+            | Less
+
+            /// !=…
+            | BangEq
+
+            /// ^…
+            | Hat
+
+            /// @…
+            | At
+
+            /// ::
+            | Cons
+
+            /// :?
+            | TypeTest
+
+            /// -…
+            | Sub
+
+            /// +…
+            | Add
+
+            /// %…
+            | Mod
+
+            /// /…
+            | Div
+
+            /// *…
+            | Mul
+
+            /// **…
+            | Exp
+
+        module InfixOperator =
+            open System
+
+            /// Tries to parse the given original notation as a symbolic infix operator.
+            let ofOriginalNotation (originalNotation: string) =
```

Comment:
Can we return value option here instead?

Reply:
Yep. I think I was originally calling this inside of `List.tryPick`, but I'm not anymore, so I'll update it.